src/Features/Core/Portable/Diagnostics/IDiagnosticService.cs

```diff
@@ -24,12 +24,12 @@ internal interface IDiagnosticService
         /// <summary>
         /// Get current diagnostics stored in IDiagnosticUpdateSource
         /// </summary>
-        ImmutableArray<DiagnosticData> GetDiagnostics(Workspace workspace, ProjectId? projectId, DocumentId? documentId, object? id, bool includeSuppressedDiagnostics, CancellationToken cancellationToken);
+        ImmutableArray<DiagnosticData> GetDiagnostics(Workspace workspace, ProjectId? projectId, DocumentId? documentId, object? id, bool includeSuppressedDiagnostics, bool forPullDiagnostics, CancellationToken cancellationToken);
 
         /// <summary>
         /// Get current buckets stored our diagnostics are grouped into.  Specific buckets can be retrieved by calling
-        /// <see cref="IDiagnosticServiceExtensions.GetDiagnostics(IDiagnosticService, DiagnosticBucket, bool, CancellationToken)"/>.
+        /// <see cref="IDiagnosticServiceExtensions.GetDiagnostics(IDiagnosticService, DiagnosticBucket, bool, bool, CancellationToken)"/>.
         /// </summary>
-        ImmutableArray<DiagnosticBucket> GetDiagnosticBuckets(Workspace workspace, ProjectId? projectId, DocumentId? documentId, CancellationToken cancellationToken);
+        ImmutableArray<DiagnosticBucket> GetDiagnosticBuckets(Workspace workspace, ProjectId? projectId, DocumentId? documentId, bool forPullDiagnostics, CancellationToken cancellationToken);
```

Comment:
Can you please add some doc comments for the new parameter?

Reply:
done.